TAScheduler/views_methods.py

```diff
@@ -193,28 +194,61 @@ def __init__(self, course_info):
         self.course_database = course_info
 
     def addInstructor(self, active_instr):
-        pass
+        if not isinstance(active_instr, InstructorObj):
+            raise TypeError("active_instr is not an instance of InstructorObj")
+        if InstructorToCourse.objects.filter(course=self.course_database).count() >= self.course_database.max_instructors:
+            raise ValueError("This course has reached the maximum number of instructors")
+        if InstructorToCourse.objects.filter(
+                instructor=active_instr.database).count() >= active_instr.database.max_assignments:
+            raise ValueError("Instructor has reached the maximum number of course assignments")
+        InstructorToCourse.objects.create(instructor=active_instr.database, course=self.course_database)
 
     def addTa(self, active_ta):
-        pass
+        if not isinstance(active_ta, TAObj):
+            raise TypeError("active_ta is not an instance of TAObj")
+        if TAToCourse.objects.filter(course=self.course_database).count() >= self.course_database.max_tas:
+            raise ValueError("This course has reached the maximum number of TAs")
+        if TAToCourse.objects.filter(ta=active_ta.database).count() >= active_ta.database.max_assignments:
+            raise ValueError("TA has reached the maximum number of course assignments")
+        TAToCourse.objects.create(ta=active_ta.database, course=self.course_database)
 
     def removeAssignment(self, active_user):
-        pass
+        # to implement a way to determine if a user is a TA or Instructor
+        if isinstance(active_user, InstructorObj):
+            InstructorToCourse.objects.filter(course=self.course_database, instructor=active_user.database).delete()
+        elif isinstance(active_user, TAObj):
+            TAToCourse.objects.filter(course=self.course_database, ta=active_user.database).delete()
+        else:
+            raise TypeError("The active_user must be an instance of InstructorObj or TAObj")
 
     def removeCourse(self):
-        pass
+        self.course_database.delete()
 
     def editCourse(self, course_info):
-        pass
+        for attr, value in course_info.items():
```

Comment:
Did brief research - seems fairly elagent and I think it will work if everything is entered correctly, but I'm unsure if this code handles cases where "not all input has been provided", or "missing input" (one of the fields are "") 
... maybe do some exception handling in the beginning of this method before doing setattr()